src/riot-rs-embassy/src/lib.rs

```diff
@@ -4,6 +4,12 @@
 
 pub mod assign_resources;
 
+#[cfg(context = "nrf52")]
```

Comment:
We can replace this with:

```
#[cfg_attr(context = "nrf52", path = "arch/nrf52.rs")]
#[cfg_attr(context = "rp2040", path = "arch/rp2040.rs")]
pub mod arch;
```

... and then drop the others. That way we only have one line per arch plus one for the "pub mod arch", instead of four per arch.